[ONNX] Unify ONNXProgram and remove the old one

justinchuby opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #136277
* #136335
* #136279
* __->__ #136281

## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now (it has a lot of xfails anyways). A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


cc @albanD

justinchuby committed:
[ONNX] Unify ONNXProgram and remove the old one

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


Fixes https://github.com/pytorch/pytorch/issues/136274


[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


Fixes https://github.com/pytorch/pytorch/issues/136274


[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


Fixes https://github.com/pytorch/pytorch/issues/136274


[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now. A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


## TODO

- Make sure onnxprogram is importable w/o onnxscript
- Update fx tests

## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now. A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


## TODO

- Make sure onnxprogram is importable w/o onnxscript
- Update fx tests

## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now. A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


## TODO

- Make sure onnxprogram is importable w/o onnxscript
- Update fx tests

## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now. A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


## TODO

- Make sure onnxprogram is importable w/o onnxscript
- Update fx tests

## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now. A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now (it has a lot of xfails anyways). A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now (it has a lot of xfails anyways). A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now (it has a lot of xfails anyways). A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


[ghstack-poisoned]

justinchuby commented:
@pytorchbot merge -i

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now (it has a lot of xfails anyways). A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


[ghstack-poisoned]

albanD reviewed (approved):
Skipping PR sanity check for code removal sounds good!
Thanks for the cleanup!

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now (it has a lot of xfails anyways). A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


cc albanD

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now (it has a lot of xfails anyways). A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


cc albanD

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now (it has a lot of xfails anyways). A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


cc albanD

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now (it has a lot of xfails anyways). A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


cc albanD

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Unify ONNXProgram and remove the old one"


## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now (it has a lot of xfails anyways). A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274


cc albanD

[ghstack-poisoned]

justinchuby commented:
@pytorchbot merge

justinchuby commented:
@pytorchbot merge -f "All relevant tests passed"

titaiwangms commented on torch/onnx/_internal/_exporter_legacy.py:
```diff
@@ -1023,13 +608,8 @@ def export(self) -> ONNXProgram:
-                self.options.diagnostic_context,
-                fake_context=self.options.fake_context,
-                model_torch=self.model,
+            return _onnx_program.ONNXProgram(
```
@justinchuby How do we call io_adapter in this scenario?
